### ISSUE TITLE
Plugin Eagerly Consumes Taskgraph

### ISSUE THREAD

**Issue by AndrewReitz** [gmazzo/gradle-buildconfig-plugin#7]
I'm adding a custom sourceSet for integration testing

```
sourceSets {
    create("integrationTest") {
        compileClasspath += sourceSets.main.get().output + configurations.testRuntimeClasspath
        runtimeClasspath += output + compileClasspath
    }
}
```

and when adding the plugin it consumes the task graph before all the tasks have been created causing this error

```
* What went wrong:
Task with name 'compileIntegrationTestKotlin' not found in project ':networking'.
```

Here's the stacktrace from the build scan. https://scans.gradle.com/s/wkhjddgpeydpk/failure#1

**Comment by dirkbolte** [comment]
I just ran into the same issue. I only added the plugin (so: no buildconfig) and had the issue right away.

**Comment by gmazzo** [comment]
As @ZacSweers correctly pointed out at https://github.com/gmazzo/gradle-buildconfig-plugin/issues/8#issuecomment-752291760, this probably will be fixed by a lazy task resolution.

I'll have to work on a deeper refactor on `PluginBindings` as it's not as straight forward than expected :(

**Comment by gmazzo** [comment]
Sorry for the late response, this should be fixed on `2.1.0`.
